Closure_Front_End/src/components/ProgressBox.vue

```diff
@@ -35,46 +35,32 @@
   </div>
 </template>
 <script>
+
+import { groupBy } from '@/utils.js'
+
+
 export default {
   props: ["allcourses"],
 
-  data() {
-    return {
-      total: 50,
-      mandatory: 30,
-      mand_choice: 20,
-      choice: 15,
-      coursesByType: null,
-      mustValue: null,
-      chooseListValue: null,
-      choiceValue: null,
-    };
-  },
-
-  methods: {
-    groupBy: function (xs, key) {
-      return xs.reduce(function (rv, x) {
-        (rv[x[key]] = rv[x[key]] || []).push(x);
-        return rv;
-      }, {});
+  computed: {
+    total() {
+      return this.allcourses.length
+    },
+    coursesByType() {
+      return groupBy(this.allcourses, "type")
+    },
+    mustValue() {
+      return 1 in this.coursesByType ? this.coursesByType[1].length : 0
+    },
+
+    chooseListValue() {
+      return 2 in this.coursesByType ? this.coursesByType[2].length : 0
     },
-  },
-
-  mounted() {
-    this.coursesByType = this.groupBy(this.allcourses, "type");
-    this.mustValue = this.coursesByType[1].length;
-    this.chooseListValue = this.coursesByType[2].length;
-    this.choiceValue = this.coursesByType[3].length;
-    console.log(this.coursesByType);
-  },
-
-  updated() {
-    this.coursesByType = this.groupBy(this.allcourses, "type");
-    this.mustValue = this.coursesByType[1].length;
-    this.chooseListValue = this.coursesByType[2].length;
-    this.choiceValue = this.coursesByType[3].length;
-    console.log(this.coursesByType);
-  },
+
+    choiceValue() {
+      return 3 in this.coursesByType ? this.coursesByType[3].length : 0

```

Comment:
What does this `3` mean? Please document shortly (code-wise, syntax is clear)

Reply:
These numbers represent types of a course. They weren't added by me, they belong to the #85 feature

I see they also appear in other places(the example courses file, and the `CourseBox` component), perhaps I will refactor them to be strings(enums) in a way that corresponds to the API model, as it doesn't really make sense to assign them a number

Reply:
I'll resolve this conversation seeing as this will be fixed in #117